components/Common/ShellBox/index.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useCallback } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { useCopyToClipboard } from '../../../hooks/useCopyToClipboard';
+import styles from './index.module.scss';
+import { createRoot } from 'react-dom/client';
+
+interface Props {
+  textToCopy?: string;
+}
+
+const ShellBox = ({ children, textToCopy }: React.PropsWithChildren<Props>) => {
+  const [copied, copyText] = useCopyToClipboard();
+
+  const handleCopyCode = useCallback(
+    async (event: React.MouseEvent<HTMLButtonElement>) => {
+      event.preventDefault();
+      const text = textToCopy
+        ? textToCopy
+        : // @ts-ignore - createRoot is not in the type definition
+          (createRoot(document.body).render(children) as string);
```

Comment:
This is not correct. ANd creating a root on `document.body` is also completely wrong.

Let's simplify things here, I think we're overdoing ourselves here. If what we want is to copy the inner content of the ShellBox, this can be accomplished easily through references:

```tsx
  // ... remaining of the code of the element

  const shellBoxRef = useRef<HTMLCodeElement>();

  // ....
  // when copying content:
  // const text = textToCopy || shellBoxRef.current?.innerHTML;

  return (
    <pre className={styles.shellBox}>
      <div className={styles.top}>
        <span>SHELL</span>
        <button type="button" onClick={handleCopyCode}>
          <FormattedMessage
            id="components.common.shellBox.copy"
            values={{ copied }}
          />
        </button>
      </div>
      <code ref={shellBoxRef}>{children}</code>
    </pre>
  );
```